src/main.rs

```
@@ -68,6 +67,16 @@ struct Config {
     #[structopt(short, long, parse(from_occurrences))]
     verbose: u64,
 
+    /// gRPC address of IOx server to connect to
+    #[structopt(
+        short,
+        long,
+        global = true,
+        env = "IOX_ADDR",
+        default_value = "http://127.0.0.1:8082"
+    )]
+    host: String, /* TODO: This must be on the root due to https://github.com/clap-rs/clap/pull/2253 */
```

To elaborate on this, in clapv2 and released versions of the clapv3 beta, global arguments registered on subcommands don't get propagated to their subcommands properly. 

This causes a confusing behaviour where if the argument is specified for the "leaf" subcommand, e.g. `influxdb_iox create database --host foo bar` it will pass validation, but the argument is silently ignored.

---

Maybe it is worth a ticket in IOx describing the issue and linking to clap so we know how to fix it when someone hits the problem (aka hope that a clap has been released that fixes this).

It is also confusing in that the help message implies that `--host` will be accepted by the sub command.

```
(arrow_dev) alamb@ip-10-0-0-49:~/Software/influxdb_iox$ target/debug/influxdb_iox database
influxdb_iox-database 0.1.0
Manage IOx databases

USAGE:
    influxdb_iox database [OPTIONS] <SUBCOMMAND>

FLAGS:
        --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -h, --host <host>    gRPC address of IOx server to connect to [env: IOX_ADDR=]  [default: http://127.0.0.1:8082]

SUBCOMMANDS:
    create    Create a new database
    get       Get list of databases, or return configuration of specific database
    help      Prints this message or the help of the given subcommand(s)
```

---

> It is also confusing in that the help message implies that --host will be accepted by the sub command.

@alamb That's the thing - it will. The bug is that it wouldn't work if host were instead registered as a global the `database` subcommand instead of the root.

With this the following are all valid

```
influxdb_iox database create foo --host HOST
influxdb_iox database create --host HOST foo
influxdb_iox database --host HOST create foo
influxdb_iox --host HOST database create foo
```

The "issue" with putting it on the root, is that this is valid but does nothing (because the server command does nothing with the argument).

```
influxdb_iox server --host foo
```